src/models/StyledComponent.js

```diff
@@ -36,9 +38,9 @@ export default (ComponentStyle) => {
             children.push(createElement('template', { slot }, this.$slots[slot]))
           }
         }
-
+        const element = this.$props.as
         return createElement(
-          target,
+          isVueComponent(target) ? target : element || target,
```

Comment:
Good point. Vue already does this under the hood.

Reply:
On second thought, I forgot why I added this check. There's two things this check helps us do:

**1) Rendering the `as` prop element provided by consumer or the default `target` element.**
In order to do this, would simple just need to do a pipe operator and we'd have it working.

```js
const StyledComponent = {
  props: {
    as: [String, Object],
    // ...
  },
  render(createElement) {
    return createElement(this.as || target)
  }
}
```
So our templates could look something like:
```vue
<template>
  <!-- Renders "div" element -->
  <Btn as="div">My button</Btn>
</template>

<script>
import styled from 'vue-styled-components'

const Btn = styled('button', { visible: Boolean })`
  border-radius: 6em;
  padding: 1em 2em;
  opacity: ${props => props.visible ? 1: 0};
}`

export default {
  components: {
    Btn
  }
}
</script>
```
This will work fine and render the `as` element as expected. 

**2) Preserving styles from lowest level component**: When passing a value into the "as" prop, 
the `target` component would end up being discarded because `this.$props.as` is defined.
Therefore, if the `target` is a Styled component with it's own styles, we would lose the styles from the lowest level Styled Component. Consider the following `CustomButton` that composes the `Btn` component from the last snippet.

```vue
<template>
  <!-- 
    Will not forward style props from `Btn` component
    because  this.$props.as === 'undefined'
  -->
  <CustomBtn as="footer" />
</template>
<script>
const CustomBtn = styled(Btn, {
  color: String,
  bg: String
})`
  color: ${props => props.color};
  background-color: ${props => props.bg};
`
export default {
  components: {
    CustomBtn
  }
}
</script>
```
This will render the `CustomBtn` as a `footer` but will not display the styles of the `Btn` component from before. In order to overcome this, we check to see if the target is indeed a Styled Component and since it renders recursively, we can then render the `as` prop element and keep the inner component styles as well as the new styles for the `CustomBtn`. So we added this check:
```js
// ...
  render(createElement) {
    return createElement(
      isVueComponent(target) ? target : this.$props.as || target
    )
  }
// ...
```